### PR TITLE
Reject non-standard Number classes during POJO serialization

### PR DESCRIPTION
**Pull request by schmidt-sebastian**
This change adds custom error messages for BigDecimal, BigInteger, AtomicInteger, etc during serialization and deserialization. 

Note that we currently do throw errors as well.
- Deserialization:
Old:  "Unknown primitive type: class java.math.BigInteger"
New: "Could not deserialize object. Deserializing to BigInteger is not supported"

- Serialization: 
Old: "Invalid data. Unsupported type: java.math.BigInteger (found in field bigInteger)"
New: "Could not serialize object. BigInteger is not supported, please use int, long, float or double"

### CODE REVIEW

File: firebase-firestore/src/main/java/com/google/firebase/firestore/util/CustomClassMapper.java [hunk 1]
```diff
@@ -119,9 +119,13 @@ public static Object convertToPlainJavaTypes(Object object) {
         throw serializeError(path, "Shorts are not supported, please use int or long");
       } else if (o instanceof Byte) {
         throw serializeError(path, "Bytes are not supported, please use int or long");
-      } else {
-        // Long, Integer, Double
+      } else if (o instanceof Long || o instanceof Integer || o instanceof Double) {
```
Comment (wilhuff) [inline firebase-firestore/src/main/java/com/google/firebase/firestore/util/CustomClassMapper.java]: The underlying UserDataConverter handles `Float` types so you can combine the cases that return a value. With this general error message it no longer seems worthwhile to handle `Short` and `Byte` separately.

Something like this:

```java
if (o instanceof Long || o instanceof Integer || o instanceof Double || o instanceof Float) {
  return o;
} else {
  throw serializeError(...);  // your new message
}
```

File: firebase-firestore/src/main/java/com/google/firebase/firestore/util/CustomClassMapper.java [hunk 2]
```diff
@@ -320,7 +324,8 @@ public static Object convertToPlainJavaTypes(Object object) {
     } else if (Character.class.isAssignableFrom(clazz) || char.class.isAssignableFrom(clazz)) {
       throw deserializeError(path, "Deserializing to chars is not supported");
     } else {
-      throw new IllegalArgumentException("Unknown primitive type: " + clazz);
+      throw deserializeError(
```
Comment (wilhuff) [inline firebase-firestore/src/main/java/com/google/firebase/firestore/util/CustomClassMapper.java]: Similarly, do we need to special case `short`, `byte`, and `char` if we give a nearly equivalent message here?

Comment (schmidt-sebastian) [reply firebase-firestore/src/main/java/com/google/firebase/firestore/util/CustomClassMapper.java]: We can probably reduce our code duplication here as well. I removed the extra checks.

File: firebase-firestore/src/main/java/com/google/firebase/firestore/util/CustomClassMapper.java (hunk 1 above)
Comment (schmidt-sebastian) [reply firebase-firestore/src/main/java/com/google/firebase/firestore/util/CustomClassMapper.java]: Yeah, that is a good suggestion. I moved some of the code duplication to the test cases instead...

**Pull request merged**